### ISSUE TITLE
Bug: Progressively enhanced anchors are flagged

### ISSUE THREAD

**Issue by jimmynotjim** [biomejs/biome#1835]
### Environment information

```bash
CLI:
  Version:                      1.5.3
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v18.12.1"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/8.19.2"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### Rule name

useValidAnchor

### Playground link

https://biomejs.dev/playground/?lintRules=all&code=YwBvAG4AcwB0ACAAcAByAG8AZwByAGUAcwBzAGkAdgBlAGwAeQBFAG4AaABhAG4AYwBlAGQARgB1AG4AYwB0AGkAbwBuACAAPQAgACgAZQApACAAPQA%2BACAAewAKACAAIABlAC4AcAByAGUAdgBlAG4AdABEAGUAZgBhAHUAbAB0ACgAKQA7AAoAIAAgAGMAbwBuAHMAdAAgAGgAcgBlAGYAIAA9ACAAZQAuAGMAdQByAHIAZQBuAHQAVABhAHIAZwBlAHQALgBnAGUAdABBAHQAdAByAGkAYgB1AHQAZQAoACcAaAByAGUAZgAnACkAOwAKAAoAIAAgAC8ALwAgAGIAaQBvAG0AZQAtAGkAZwBuAG8AcgBlACAAbABpAG4AdAA6AG4AbwBDAG8AbgBzAG8AbABlAEwAbwBnADoAIABKAHUAcwB0ACAAZgBvAHIAIAB0AGUAcwB0AGkAbgBnAAoAIAAgAGMAbwBuAHMAbwBsAGUALgBsAG8AZwAoAGUALgBjAHUAcgByAGUAbgB0AFQAYQByAGcAZQB0AC4AZwBlAHQAQQB0AHQAcgBpAGIAdQB0AGUAKAAnAGgAcgBlAGYAJwApACkACgAgACAAYQBsAGUAcgB0ACgAYABZAG8AdQAgAGEAcgBlACAAbgBhAHYAaQBnAGEAdABpAG4AZwAgAHQAbwAgACIAIgAkAHsAaAByAGUAZgB9ACIAIgBgACkAOwAKACAAIAB3AGkAbgBkAG8AdwAuAGwAbwBjAGEAdABpAG8AbgAuAGgAcgBlAGYAIAA9ACAAaAByAGUAZgA7AAoAfQAKAAoACgAgACAAPABhACAACgAgACAAIAAgAGgAcgBlAGYAPQAiAGgAdAB0AHAAcwA6AC8ALwBnAGkAdABoAHUAYgAuAGMAbwBtACIACgAgACAAIAAgAG8AbgBDAGwAaQBjAGsAPQB7AHAAcgBvAGcAcgBlAHMAcwBpAHYAZQBsAHkARQBuAGgAYQBuAGMAZQBkAEYAdQBuAGMAdABpAG8AbgB9AAoAIAAgAD4ACgAgACAAIAAgAFQAaABpAHMAIABpAHMAIABhACAAcAByAG8AZwByAGUAcwBzAGkAdgBlAGwAeQAgAGUAbgBoAGEAbgBjAGUAZAAgAGwAaQBuAGsACgAgACAAPAAvAGEAPgA%3D

### Expected result

Adding an onClick property should not throw an error when there is a valid href value. Deque themselves show an example of a link progressively enhanced on their documentation for the `href-no-hash rule` https://dequeuniversity.com/rules/axe/html/4.3/href-no-hash

<img width="969" alt="Screenshot 2024-02-15 at 5 08 04 PM" src="https://github.com/biomejs/biome/assets/1280430/5df4a53e-af61-41ce-ad89-06b703bef2b2">

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by XiNiHa** [comment]
This also applies to cases where we just want to track the clicks of anchors, mainly due to a marketing purpose.
